### DISCUSSION TITLE
How does Gitbook handle mono-repos?

### DISCUSSION THREAD

**Discussion by rijuchatterjee** [GitbookIO/community#465] [3. Git Sync]
### Ask a question, start a conversation, or make an annoucement

How does gitbook handles monorepos? I have a folder where multiple teams are pushing their code in their respective folders. How can this be managed?

**Comment by rodrcastro** [comment]
Hey @rijuchatterjee 👋 

We have a really great resource in our docs explaining how monorepos work and how we support them with Git Sync. [You can read more about it here.](https://docs.gitbook.com/integrations/git-sync/monorepos).

If you have any questions about it, feel free to send us an email at support@gitbook.com

**Reply by rijuchatterjee** [reply]
Thanks @rodrcastro. I tried to implement the github to gitbook sync but I ran into some issues. Below is the structure of my monorepo:

`/monorepo`
``  product-team1``
```    documents.txt```
``  product-team2``
```    documents.txt```

Can you help me in configuring the sync for the above directory structure? I have setup the authorization. I just need help in what/how to update the "project directory", also with the location and the contents of .gitbook.yaml. Moreover, I wanted to know how to ignore some files for which we do not want the sync?

**Reply by rodrcastro** [reply]
Hey @rijuchatterjee. As detailed in the guide, all you have to do is set the folder you'd like to sync from the repo as the project directory.

In your example, in the space you'd like to sync with the folder `product-team1`, you set the project directoy as:

`./product-team1`

GitBook will figure out all of the configuration from there, you don't need to add a .gitbook.yaml file if you don't have use for it. GitBook will also automatically create the Readme and Summary.md files if you don't have them in that folder. If you'd like to add one for each project in the future, simply add it to the root of the directory that is synced. In the example above, you'd put the .gitbook.yaml file under the folder `product-team1`:

```
./
  product-team1
    page-1.md
    README.md
    SUMMARY.md
   .gitbook.yaml
```

Please note that only Markdown files are supported with Git Sync.

**Reply by rijuchatterjee** [reply]
HI @rodrcastro Is it possible to sync all the "product" folders present in the "monorepo" directory in one single space? So that the documents inside the folder also appear in the space in different folders?

**Reply by rodrcastro** [reply]
@rijuchatterjee No, this is not possible. Git Sync can only sync one repo/folder per space. If you want to sync multiple repos/folders, they must be separated in multiple spaces.

**Reply by zdevaty** [reply]
> Git Sync can only sync one repo/folder per space.

@rodrcastro I have a use case for opposite of monorepo - microservices/apps. In one space, I would like to sync many repos (a repo per app). Do you plan to support this sometime in the future?

**Reply by rodrcastro** [reply]
@zdevaty We don't have plans to add that to Git Sync, it'll continue to be a single repo per space for the foreseeable future.

**Reply by zdevaty** [reply]
Okay, thank you for clarifying that.
